relays/sonof_mini_r2/firmware.lua

```diff
@@ -0,0 +1,300 @@
+json = require("json")
+
+-- Configuration variables must be also defined
+-- in `write_configuration` command arguments in manifest.yml
+IP_ADDRESS = 'ip_address'
+PORT = 'port'
+
+ip_address = nil
+port = nil
+client = nil
+
+-- Initiate device firmware. Called at the end of the file.
+function main()
+  scheduler.add(30000, send_properties)
+  scheduler.add(1000, send_telemetry)
+
+  enapter.register_command_handler('activate_switch', command_switch)
+  config.init({
+    [IP_ADDRESS] = {type = 'string', required = true},
+    [PORT] = {type = 'string', required = true}
+  })
+end
+
+function send_properties()
+  local sonoff, err  = connect_sonoff()
+  if err then
+    enapter.log("Can't connect to Sonoff: "..err)
+    return
+  else
+    ip_address = sonoff.ip_address
+    port = sonoff.port
+    client = sonoff.client
+
+    local snf_data = sonoff:get_device_info()
+    if next(snf_data) then
+      enapter.send_properties({
+      vendor = 'Sonoff',
+      model = 'MINI R2',
+      fw_version = snf_data['data']['fwVersion'],
+      ip_address = ip_address,
+      port = port
+    })
+    end
+  end
+end
+
+
+function send_telemetry()
+  local sonoff, err  = connect_sonoff()
+  local alerts = {}
+  if err then
+    enapter.log("Can't connect to Sonoff: "..err)
+    enapter.send_telemetry({
+    connection_status = 'no_data',
+      alerts = {'connection_err'}
+    })
+    return
+  else
+    local snf_data = sonoff:get_device_info()
+    if snf_data ~= nil then
+      local telemetry = {}
+      local status -- та же песня,unused variable status
+      telemetry.status = pretty_status(snf_data["data"]["switch"])
+      telemetry.signal = snf_data['data']['signalStrength']
+      telemetry.connection_status = 'ok'
+      enapter.send_telemetry(telemetry)
+    else
+      enapter.send_telemetry({status = 'no_data', connection_status = 'error', alerts = {'no_data'}})
+    end
+  end
+end
+
+function pretty_status(switch_state)
+  if switch_state == 'on' then
+    return 'switch_on'
+  else if switch_state == 'off' then
+    return 'switch_off'
+  else
+    enapter.log("Can't read device state "..err) -- ругается на err, может поменять на ctx.error?
```

Comment:
There is no `err` variable in `pretty_status`, that's why you can't use it here.
just leave:
`enapter.log("Unknown device state ", 'error')`
`return switch_state`